### PR TITLE
hook up memory manager to free

### PR DESCRIPTION
**Pull request by cgardens**

### CODE REVIEW

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java [hunk 1]
```diff
@@ -17,25 +17,24 @@
...
-  BufferManagerDequeue bufferManagerDequeue;
+  private final Map<StreamDescriptor, MemoryBoundedLinkedBlockingQueue<AirbyteMessage>> buffers = new HashMap<>();
+  private final BufferManagerEnqueue bufferManagerEnqueue;
+  private final BufferManagerDequeue bufferManagerDequeue;
   private final ScheduledExecutorService debugLoop = Executors.newSingleThreadScheduledExecutor();
 
   public BufferManager() {
-    buffers = new HashMap<>();
-    final var memoryManager = new GlobalMemoryManager(TOTAL_QUEUES_MAX_SIZE_LIMIT_BYTES);
+    final long availableMemory = (long) (Runtime.getRuntime().maxMemory() * 0.8);
```
Comment (davinchia) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:35]: can use Line 23.

Comment (cgardens) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java]: thanks.

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java [hunk 2]
```diff
@@ -202,33 +196,37 @@ public void close() throws Exception {
...
+      final var freeMem = maxMemoryBytes - currentMemoryBytes.get();
       // Never allocate more than free memory size.
       final var toAllocateBytes = Math.min(freeMem, BLOCK_SIZE_BYTES);
-      currentMemoryBytes += toAllocateBytes;
+      currentMemoryBytes.addAndGet(toAllocateBytes);
 
       return toAllocateBytes;
     }
 
     public void free(final long bytes) {
```
Comment (davinchia) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]:
```suggestion
    public void synchronized free(final long bytes) {
```

Comment (davinchia) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]: what calls 'free'?

Comment (cgardens) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]: why does this need to be synchronized? i get it for request memory. but not for free.

Comment (davinchia) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]: wait figured out what calls free

Comment (davinchia) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]: I believe free is called in each worker pool thread

Comment (cgardens) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]: yeah, but it is thread safe. it's just updating an atomiclong. there's no reason why 2 threads can't do that at the same time.

Comment (davinchia) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java:225]: you are right

**Review by davinchia — APPROVED** [review]

**Pull request merged**